ccloud/resource_api_key.go

```diff
@@ -173,8 +174,13 @@ func apiKeyDelete(ctx context.Context, d *schema.ResourceData, meta interface{})
 	}
 
 	id := d.Id()
-	log.Printf("[INFO] Deleting API key %s in account %s", id, accountID)
-	err := c.DeleteAPIKey(id, accountID, logicalClustersReq)
+	serviceAccount, _ := c.ReadServiceAccount(userID)
```

Comment:
I would check if `c.ReadServiceAccount` returns `error` and if yes return `diag.FromErr(err)`.